Index: CMakeModules/FindOpenEXR.cmake
===================================================================
--- CMakeModules/FindOpenEXR.cmake	(revision 2980)
+++ CMakeModules/FindOpenEXR.cmake	(working copy)
@@ -22,6 +22,12 @@
   INCLUDE(UsePkgConfig)
   
   PKGCONFIG(OpenEXR _OpenEXRIncDir _OpenEXRLinkDir _OpenEXRLinkFlags _OpenEXRCflags)
+  
+  # ugly hack, sets BAD_OPENEXR varible when OpenEXR <= 1.5.0 is used
+  # because these old versions doesn't set -lpthread flag correcly
+  pkg_check_modules(OPENEXR OpenEXR)
+  STRING(COMPARE GREATER OPENEXR__VERSION 1.5.0 BAD_OPENEXR)
+  
 ENDIF (NOT WIN32 OR MINGW)
   FIND_PATH(OPENEXR_INCLUDE_DIR ImfRgbaFile.h
      ${_OpenEXRIncDir}
@@ -134,6 +140,10 @@
        else (MSVC)
          set(OPENEXR_LIBRARIES ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_HALF_LIBRARY} ${OPENEXR_ILMTHREAD_LIBRARY})
        endif (MSVC)
+# if BAD_OPENEXR is set, add -lpthread flag manually
+       if (BAD_OPENEXR)
+         set(OPENEXR_LIBRARIES ${OPENEXR_LIBRARIES} -lpthread)
+       endif (BAD_OPENEXR)
      else (OPENEXR_ILMTHREAD_LIBRARY)
        if (MSVC)
          set(OPENEXR_LIBRARIES optimized ${OPENEXR_IMATH_LIBRARY} optimized ${OPENEXR_ILMIMF_LIBRARY} optimized ${OPENEXR_IEX_LIBRARY} optimized ${OPENEXR_HALF_LIBRARY}  debug ${OPENEXR_IMATH_LIBRARY_DEBUG} debug ${OPENEXR_ILMIMF_LIBRARY_DEBUG} debug ${OPENEXR_IEX_LIBRARY_DEBUG} debug ${OPENEXR_HALF_LIBRARY_DEBUG} CACHE STRING "The libraries needed to use OpenEXR")
