Index: src/matchpoint/CMakeLists.txt
===================================================================
--- src/matchpoint/CMakeLists.txt	(revision 2980)
+++ src/matchpoint/CMakeLists.txt	(working copy)
@@ -1,6 +1,11 @@
 
 #list all source files here
 ADD_EXECUTABLE(matchpoint MatchPoint.cpp APImage.cpp HessianDetector.cpp Descriptor.cpp)
+
+IF (UNIX)
+	SET_TARGET_PROPERTIES(matchpoint PROPERTIES LINK_FLAGS -lpthread)
+ENDIF (UNIX)
+
 IF (MSVC)
 	TARGET_LINK_LIBRARIES(matchpoint ${image_libs} hugingetopt)
 #need to link to some other libraries ? just add them here
