--- libpqxx-2.6.9/configure.orig	2007-02-01 13:10:03.000000000 +0300
+++ libpqxx-2.6.9/configure	2007-11-19 00:25:17.489832798 +0300
@@ -19633,68 +19633,7 @@
 		add_compiler_opts -Wno-long-long
 	fi
 
-{ echo "$as_me:$LINENO: checking g++ visibility attribute" >&5
-echo $ECHO_N "checking g++ visibility attribute... $ECHO_C" >&6; }
-gcc_visibility=yes
-SAVE_CXXFLAGS="$CXXFLAGS"
-CXXFLAGS="$CXXFLAGS -Werror"
-cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-
-struct __attribute__ ((visibility("hidden"))) d { d() {} void f() {} };
-
-int
-main ()
-{
-d D;D.f()
-  ;
-  return 0;
-}
-_ACEOF
-rm -f conftest.$ac_objext
-if { (ac_try="$ac_compile"
-case "(($ac_try" in
-  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
-  *) ac_try_echo=$ac_try;;
-esac
-eval "echo \"\$as_me:$LINENO: $ac_try_echo\"") >&5
-  (eval "$ac_compile") 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } && {
-	 test -z "$ac_cxx_werror_flag" ||
-	 test ! -s conftest.err
-       } && test -s conftest.$ac_objext; then
-
-cat >>confdefs.h <<\_ACEOF
-#define PQXX_HAVE_GCC_VISIBILITY 1
-_ACEOF
-
-else
-  echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-	gcc_visibility=no
-fi
-
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-{ echo "$as_me:$LINENO: result: $gcc_visibility" >&5
-echo "${ECHO_T}$gcc_visibility" >&6; }
-	if test "$gcc_visibility" = "yes" ; then
-		# Setting default symbol visibility to "hidden" vastly reduces
-		# library size with g++ 4.0, but unfortunately this will require
-		# some more work in the libpqxx setup.  Instead, we make only
-		# inline functions hidden.
-		add_compiler_opts -fvisibility-inlines-hidden
-		#add_compiler_opts -fvisibility=hidden
-	fi
+gcc_visibility=no
 CXXFLAGS="$SAVE_CXXFLAGS"
 
 { echo "$as_me:$LINENO: checking g++ deprecation attribute" >&5
