--- libofa-0.9.3/lib/JAMA/tnt_math_utils.h.orig	2007-11-09 15:37:55.000000000 -0500
+++ libofa-0.9.3/lib/JAMA/tnt_math_utils.h	2007-11-09 15:38:16.000000000 -0500
@@ -19,6 +19,12 @@
 
 namespace TNT
 {
+template <class Real>
+Real abs(const Real &a)
+{
+	return  (a > 0 ? a : -a);
+}
+
 /**
 	@returns hypotenuse of real (non-complex) scalars a and b by 
 	avoiding underflow/overflow
@@ -59,12 +65,6 @@
 /**
 	@returns the absolute value of a real (no-complex) scalar.
 */
-template <class Real>
-Real abs(const Real &a)
-{
-	return  (a > 0 ? a : -a);
-}
-
 }
 #endif
 /* MATH_UTILS_H */
