diff -r 7b56deeaef2e src/Notification.cs
--- a/src/Notification.cs	Fri Feb 20 15:25:37 2009 -0500
+++ b/src/Notification.cs	Fri Feb 20 15:51:35 2009 -0500
@@ -273,10 +273,13 @@
 		}
 
 		public void AddAction (string action, string label, ActionHandler handler) {
-			lock (action_map) {
-				action_map[action] = new ActionTuple (label, handler);
+                       if (Notifications.Global.Capabilities != null &&
+                           Array.IndexOf (Notifications.Global.Capabilities, "actions") > -1) {
+                               lock (action_map) {
+                                       action_map[action] = new ActionTuple (label, handler);
+                               }
+                               Update ();
 			}
-			Update ();
 		}
 
 		public void RemoveAction (string action) {
