diff -r 0ac05c73c8ee main-menu/src/network-status-agent.c
--- a/main-menu/src/network-status-agent.c	Fri Aug 01 13:56:36 2008 -0400
+++ b/main-menu/src/network-status-agent.c	Fri Aug 01 19:42:33 2008 -0400
@@ -23,8 +23,8 @@
 #include <string.h>
 #include <libnm-glib/nm-client.h>
 #include <NetworkManager.h>
-#include <libnm-glib/nm-device-802-11-wireless.h>
-#include <libnm-glib/nm-device-802-3-ethernet.h>
+#include <libnm-glib/nm-device-wifi.h>
+#include <libnm-glib/nm-device-ethernet.h>
 #include <nm-setting-ip4-config.h>
 #include <nm-utils.h>
 #include <arpa/inet.h>
@@ -213,13 +213,13 @@
 	def_addr = nm_ip4_config_get_addresses (cfg);
 	if (def_addr) {
 		info->ip4_addr = ip4_address_as_string (def_addr->address);
-		info->subnet_mask = ip4_address_as_string (def_addr->netmask);
+		/* info->subnet_mask = ip4_address_as_string (def_addr->netmask); */
 		info->route = ip4_address_as_string (def_addr->gateway);
 
-		network = ntohl (def_addr->address) & ntohl (def_addr->netmask);
-		hostmask = ~ntohl (def_addr->netmask);
-		bcast = htonl (network | hostmask);
-		info->broadcast = ip4_address_as_string (bcast);
+		/* network = ntohl (def_addr->address) & ntohl (def_addr->netmask); */
+		/* hostmask = ~ntohl (def_addr->netmask); */
+		/* bcast = htonl (network | hostmask); */
+		/* info->broadcast = ip4_address_as_string (bcast); */
 	}
 
 	info->primary_dns = NULL;
@@ -233,16 +233,16 @@
 			info->secondary_dns = ip4_address_as_string (g_array_index (array, guint32, 1));
 	}
 
-	if (NM_IS_DEVICE_802_11_WIRELESS(device))
+	if (NM_IS_DEVICE_WIFI(device))
 	{
 		NMAccessPoint * activeap = NULL;
 		const GByteArray * ssid;
 
 		info->type = DEVICE_TYPE_802_11_WIRELESS;
-		info->speed_mbs = nm_device_802_11_wireless_get_bitrate (NM_DEVICE_802_11_WIRELESS(device));
-		info->hw_addr = g_strdup (nm_device_802_11_wireless_get_hw_address (NM_DEVICE_802_11_WIRELESS(device)));
+		info->speed_mbs = nm_device_wifi_get_bitrate (NM_DEVICE_WIFI(device));
+		info->hw_addr = g_strdup (nm_device_wifi_get_hw_address (NM_DEVICE_WIFI(device)));
 
-		activeap = nm_device_802_11_wireless_get_active_access_point (NM_DEVICE_802_11_WIRELESS(device));
+		activeap = nm_device_wifi_get_active_access_point (NM_DEVICE_WIFI(device));
 		if (activeap)
 		{
 			ssid = nm_access_point_get_ssid (NM_ACCESS_POINT (activeap));
@@ -253,11 +253,11 @@
 		if (! info->essid)
 			info->essid = g_strdup ("(none)");
 	}
-	else if (NM_IS_DEVICE_802_3_ETHERNET (device))
+	else if (NM_IS_DEVICE_ETHERNET (device))
 	{
 		info->type = DEVICE_TYPE_802_3_ETHERNET;
-		info->speed_mbs = nm_device_802_3_ethernet_get_speed (NM_DEVICE_802_3_ETHERNET(device));
-		info->hw_addr = g_strdup (nm_device_802_3_ethernet_get_hw_address (NM_DEVICE_802_3_ETHERNET(device)));
+		info->speed_mbs = nm_device_ethernet_get_speed (NM_DEVICE_ETHERNET(device));
+		info->hw_addr = g_strdup (nm_device_ethernet_get_hw_address (NM_DEVICE_ETHERNET(device)));
 	}
 
 	return info;
