diff -rupN qsynth-0.3.6.old/CMakeLists.txt qsynth-0.3.6/CMakeLists.txt
--- qsynth-0.3.6.old/CMakeLists.txt	2011-04-02 08:39:39.000000000 -0400
+++ qsynth-0.3.6/CMakeLists.txt	2011-04-09 09:41:02.000000000 -0400
@@ -50,6 +50,9 @@ endif (WIN32)
 # Check for fluidsynth library.
 find_library ( FLUIDSYNTH_LIBRARY fluidsynth )
 
+# Check for X11 library.
+find_library ( X11_LIBRARY X11 )
+
 # Checks for header files.
 check_include_files ("fcntl.h;unistd.h;signal.h" HAVE_SIGNAL_H)
 
diff -rupN qsynth-0.3.6.old/src/CMakeLists.txt qsynth-0.3.6/src/CMakeLists.txt
--- qsynth-0.3.6.old/src/CMakeLists.txt	2011-04-09 09:31:32.000000000 -0400
+++ qsynth-0.3.6/src/CMakeLists.txt	2011-04-09 09:41:27.000000000 -0400
@@ -110,6 +110,7 @@ target_link_libraries ( qsynth
     ${QT_LIBRARIES}
     ${MATH_LIBRARY}
     ${FLUIDSYNTH_LIBRARY}
+    ${X11_LIBRARY}
 )
 
 set ( TRANSLATIONS
