diff -r 355a9cb4b39b data/Makefile.am
--- a/data/Makefile.am	Mon Apr 20 16:52:23 2009 -0400
+++ b/data/Makefile.am	Mon Apr 20 16:52:43 2009 -0400
@@ -7,7 +7,7 @@
 	logout-dialog.glade \
 	logout-dialog.xml \
 	new.svg \
-	ubuntu.png \
+	distro-logo.png \
 	scrollbar.svg \
 	shortcutbg.svg \
 	shortcuthilight.svg \
diff -r 355a9cb4b39b src/launcher-app.c
--- a/src/launcher-app.c	Mon Apr 20 16:52:23 2009 -0400
+++ b/src/launcher-app.c	Mon Apr 20 16:52:43 2009 -0400
@@ -601,8 +601,8 @@
                               PANEL_HEIGHT + 5);
   clutter_actor_show (priv->window);
 
-  /* Ubuntu logo */
-  priv->ubuntu = launcher_util_texture_new_from_file (PKGDATADIR"/ubuntu.png");
+  /* Distro logo */
+  priv->ubuntu = launcher_util_texture_new_from_file ("/usr/share/pixmaps/ume-distro-logo.png");
   clutter_container_add_actor (CLUTTER_CONTAINER (stage), priv->ubuntu);
   clutter_actor_set_anchor_point_from_gravity (priv->ubuntu, 
                                                CLUTTER_GRAVITY_NORTH);
