diff -r 355a9cb4b39b src/launcher-sidebar.c
--- a/src/launcher-sidebar.c	Mon Apr 20 16:52:23 2009 -0400
+++ b/src/launcher-sidebar.c	Mon Apr 20 16:57:41 2009 -0400
@@ -1087,6 +1087,7 @@
   return shortcut;
 }
 
+/*
 static void 
 on_quit_clicked (ClutterActor *shortcut, gpointer null)
 {
@@ -1098,6 +1099,7 @@
 
   gtk_widget_destroy (quit);
 }
+*/
 
 static void
 load_volumes (LauncherSidebar *sidebar)
@@ -1150,10 +1152,13 @@
   clutter_container_add_actor (CLUTTER_CONTAINER (sidebar), shortcut);
   clutter_actor_set_position (shortcut, 0, 
                               cfg->win_height - (bar_height)-(PANEL_HEIGHT*2));
-  clutter_actor_set_name (shortcut, "gnome-session-save --gui --kill");
+  clutter_actor_set_name (shortcut, "gnome-shutdown");
   clutter_actor_show_all (shortcut);
   priv->links = g_list_append (priv->links, shortcut);
+  /*
   g_signal_connect (shortcut, "clicked", G_CALLBACK (on_quit_clicked), NULL);
+  */
+  g_signal_connect (shortcut, "clicked", G_CALLBACK (on_link_clicked), NULL);
 }
 
 static void
