diff -Naur pulseaudio-0.9.8.orig/src/daemon/PulseAudio.policy pulseaudio-0.9.8/src/daemon/PulseAudio.policy
--- pulseaudio-0.9.8.orig/src/daemon/PulseAudio.policy	2007-11-16 16:42:19.000000000 -0500
+++ pulseaudio-0.9.8/src/daemon/PulseAudio.policy	2008-01-16 08:38:38.000000000 -0500
@@ -32,7 +32,7 @@
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
-      <allow_active>auth_admin_keep_always</allow_active>
+      <allow_active>yes</allow_active>
     </defaults>
   </action>
 
@@ -42,7 +42,7 @@
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
-      <allow_active>auth_admin_keep_always</allow_active>
+      <allow_active>yes</allow_active>
     </defaults>
   </action>
 
