--- mkinitrd.orig	2010-05-14 21:50:13.000000000 -0400
+++ mkinitrd	2010-05-14 21:53:23.000000000 -0400
@@ -1721,6 +1721,27 @@
     emitcrypto `eval echo '$'$cryptdev`
 done
 
+if [ -x /sbin/libata-migrate-links ]; then
+    /sbin/libata-migrate-links
+    for i in /boot/grub/grub.conf /etc/bootloader.conf /etc/fstab \
+         /etc/smartd.conf /boot/grub/device.map /etc/mtools.conf $RCFILE
+    do  
+        [ -f $i ] && DEVSUB="$DEVSUB $i" 
+    done
+    rm -f /etc/sysconfig/devsub # or it may not modify $RCFILE
+    [ -n "$DEVSUB" ] && /usr/bin/devsub /etc/sysconfig $DEVSUB
+    rm -f /etc/blkid.tab /etc/lvm/cache/.cache
+
+    inst /bin/ash.static "$MNTIMAGE" /bin/ash
+    ln -s ash "$MNTIMAGE"/bin/bash
+    for i in /sbin/libata-migrate-links /bin/ln /bin/mkdir /bin/basename \
+         /bin/cat /bin/sed /sbin/*.libata-migrate
+    do
+        inst $i "$MNTIMAGE"
+    done
+    emit "/sbin/libata-migrate-links"
+fi
+
 if [ -z "$noresume" -a -n "$thawdev" ]; then
     emit "resume $thawdev"
 fi
