diff -r 191ea3f6bcb5 wxPython/setup.py
--- a/wxPython/setup.py	Sat Oct 23 19:01:34 2010 -0500
+++ b/wxPython/setup.py	Sun Oct 24 00:04:30 2010 -0500
@@ -926,28 +926,11 @@
         SCRIPTS.append( opj('scripts/genaxmodule') ) 
 
 
-
+SCRIPTS = None
 DATA_FILES += find_data_files('wx/lib/editor', '*.txt')
 DATA_FILES += find_data_files('wx/py', '*.txt', '*.ico', '*.css', '*.html')
 
-DATA_FILES += find_data_files('wx/tools/XRCed', '*.txt', '*.xrc', '*.htb')
-DATA_FILES += find_data_files('wx/tools/XRCed/plugins', '*.crx')
-DATA_FILES += find_data_files('wx/tools/XRCed/plugins/bitmaps', '*.png')
 
-DATA_FILES += find_data_files('wx/tools/Editra/docs', '*.txt')
-DATA_FILES += find_data_files('wx/tools/Editra/locale', '*.mo')
-DATA_FILES += find_data_files('wx/tools/Editra/pixmaps',
-                              '*.png', '*.icns', '*.ico', 'README', 'AUTHORS', 'COPYING')
-DATA_FILES += find_data_files('wx/tools/Editra/plugins', '*.egg')
-DATA_FILES += find_data_files('wx/tools/Editra/src', 'README')
-DATA_FILES += find_data_files('wx/tools/Editra/styles', '*.ess')
-DATA_FILES += find_data_files('wx/tools/Editra/tests/syntax', '*')
-DATA_FILES += find_data_files('wx/tools/Editra', '[A-Z]*', recursive=False)
-
-
-## import pprint
-## pprint.pprint(DATA_FILES)
-## sys.exit()
 
 
 if NO_HEADERS or EGGing:
