diff -r af41af583ecc configure
--- a/configure	Tue Feb 07 13:48:04 2012 +0100
+++ b/configure	Tue Feb 07 13:50:08 2012 +0100
@@ -11785,7 +11785,7 @@
     PKG_CONFIG=no
   fi
 
-  min_glib_version=2.30.0
+  min_glib_version=2.28.0
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for GLIB - version >= $min_glib_version" >&5
 $as_echo_n "checking for GLIB - version >= $min_glib_version... " >&6; }
 
