--- i7z_rw_registers.rb.orig	2011-11-14 11:53:53.321414835 +0000
+++ i7z_rw_registers.rb	2011-11-14 11:54:07.212414500 +0000
@@ -1,3 +1,5 @@
+#!/usr/bin/ruby
+
 #* ----------------------------------------------------------------------- *
 # *
 # *   Under GPL v3
@@ -6,8 +8,6 @@
 # *
 # * ----------------------------------------------------------------------- */
 
-#!/usr/bin/ruby
-
 def print_command_list()
 	print "Do you need help? \n"
 	print "Possible commands are \n" 
