--- ntop-3.4-pre3.org/Makefile.am	2010-02-13 17:08:02.000000000 +0530
+++ ntop-3.4-pre3/Makefile.am	2010-05-04 13:23:12.291539308 +0530
@@ -336,7 +336,7 @@
 install-selinux-policy:
 	chcon -t textrel_shlib_t $(DESTDIR)$(libdir)/*ntop*.so
 
-install-data-local: ntop.txt ntop.html faq.html dnetter
+install-data-local: ntop.txt ntop.html faq.html
 	@$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(libdir)/plugins;
 	@if ! test -d $(DESTDIR)/$(datadir)/ntop; then \
 	  $(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/ntop; \
