--- metacity-2.4.13.90/src/metacity.schemas.in.ctrlaltdel	2003-01-16 15:11:47.000000000 -0500
+++ metacity-2.4.13.90/src/metacity.schemas.in	2003-01-16 15:13:41.000000000 -0500
@@ -1549,8 +1549,31 @@
     </schema>
 
     <schema>
-      <key>/schemas/apps/metacity/global_keybindings/run_command</key>
+      <key>/schemas/apps/metacity/global_keybindings/run_command_1</key>
       <applyto>/apps/metacity/global_keybindings/run_command_1</applyto>
+      <owner>metacity</owner>
+      <type>string</type>
+      <default>&lt;Ctrl&gt;&lt;Alt&gt;Delete</default>
+      <locale name="C">
+         <short>Run a defined command</short>
+         <long>
+          The keybinding that runs the correspondingly-numbered
+          command in /apps/metacity/keybinding_commands
+
+          The format looks like "&lt;Control&gt;a" or
+          "&lt;Shift&gt;&lt;Alt&gt;F1.
+
+          The parser is fairly liberal and allows lower or upper case,
+          and also abbreviations such as "&lt;Ctl&gt;" and
+          "&lt;Ctrl&gt;". If you set the option to the special string
+          "disabled", then there will be no keybinding for this
+          action.
+         </long>
+      </locale>
+    </schema>
+
+    <schema>
+      <key>/schemas/apps/metacity/global_keybindings/run_command</key>
       <applyto>/apps/metacity/global_keybindings/run_command_2</applyto>
       <applyto>/apps/metacity/global_keybindings/run_command_3</applyto>
       <applyto>/apps/metacity/global_keybindings/run_command_4</applyto>
@@ -1586,8 +1609,24 @@
     <!-- commands to run with the run_command keybindings -->
 
     <schema>
-      <key>/schemas/apps/metacity/keybinding_commands/command</key>
+      <key>/schemas/apps/metacity/keybinding_commands/command_1</key>
       <applyto>/apps/metacity/keybinding_commands/command_1</applyto>
+      <owner>metacity</owner>
+      <type>string</type>
+      <default>gnome-session-save --kill</default>
+      <locale name="C">
+         <short>Commands to run in response to keybindings</short>
+         <long>
+         The /apps/metacity/global_keybindings/run_command_N 
+         keys define keybindings that correspond to these commands.
+         Pressing the keybinding for run_command_N will 
+         execute command_N.
+         </long>
+      </locale>
+    </schema>
+
+    <schema>
+      <key>/schemas/apps/metacity/keybinding_commands/command</key>
       <applyto>/apps/metacity/keybinding_commands/command_2</applyto>
       <applyto>/apps/metacity/keybinding_commands/command_3</applyto>
       <applyto>/apps/metacity/keybinding_commands/command_4</applyto>
