diff --git a/modules/Makefile.in b/modules/Makefile.in
--- a/modules/Makefile.in
+++ b/modules/Makefile.in
@@ -61,7 +61,7 @@
 MODULE_INSTALL_DIR = @libdir@/slang/v@slang_major_version@/modules
 INSTALL		= @INSTALL@
 INSTALL_DATA	= @INSTALL_DATA@
-INSTALL_MODULE	= @INSTALL_MODULE@
+INSTALL_MODULE	= @INSTALL@
 MKINSDIR	= $(CONFIG_DIR)/autoconf/mkinsdir.sh
 SLSH_DIR	= $(prefix)/share/slsh
 CMAPS_DIR	= $(SLSH_DIR)/cmaps
@@ -75,7 +75,7 @@
 DEST_CMAPSDIR	= $(DESTDIR)$(CMAPS_DIR)
 DEST_HELPDIR	= $(DESTDIR)$(HELP_DIR)
 #---------------------------------------------------------------------------
-RPATH = @RPATH@
+RPATH = 
 
 LIBS = $(LDFLAGS) $(RPATH) $(SLANG_LIB) $(DL_LIB) @M_LIB@
 INCS = $(SLANG_SRCINC)
diff --git a/slsh/Makefile.in b/slsh/Makefile.in
--- a/slsh/Makefile.in
+++ b/slsh/Makefile.in
@@ -19,7 +19,7 @@
 SLANG_SRCLIB	= -L@ELFDIR@
 #---------------------------------------------------------------------------
 OTHER_LIBS	= @TERMCAP@ @DYNAMIC_LINK_LIB@ @LIBS@ @M_LIB@
-RPATH		= @RPATH@
+RPATH		= 
 #----------------------------------------------------------------------------
 INSTALL		= @INSTALL@
 INSTALL_DATA	= @INSTALL_DATA@
diff --git a/src/Makefile.in b/src/Makefile.in
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -195,7 +195,7 @@
 	-$(RM) $(DEST_LIBDIR)/$(ELFLIB)
 	-$(RM) $(DEST_LIBDIR)/$(ELFLIB_MAJOR)
 	@echo installing $(ELFLIB_BUILD_NAME) in $(DEST_LIBDIR)
-	$(INSTALL_DATA) $(ELFDIR_ELF_LIB) $(DEST_LIBDIR)/
+	$(INSTALL) $(ELFDIR_ELF_LIB) $(DEST_LIBDIR)/
 	-cd $(DEST_LIBDIR) && $(LN) $(ELFLIB_BUILD_NAME) $(ELFLIB_MAJOR)
 	-cd $(DEST_LIBDIR) && $(LN) $(ELFLIB_MAJOR) $(ELFLIB)
 	@echo ""
