commit dff93c53664cde0fa5f3cfd44a939ca1d9d090af
Author: Mark Trompell <mark@foresightlinux.org>
Date:   Fri Jan 4 08:31:02 2013 +0100

    tweak helpers

diff --git a/exo-helper/helpers/Terminal.desktop.in.in b/exo-helper/helpers/Terminal.desktop.in.in
index 1979f59..1b10ee0 100644
--- a/exo-helper/helpers/Terminal.desktop.in.in
+++ b/exo-helper/helpers/Terminal.desktop.in.in
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Version=1.0
-Icon=Terminal
+Icon=utilities-terminal
 Type=X-XFCE-Helper
 _Name=Xfce Terminal
 StartupNotify=true
-X-XFCE-Binaries=Terminal;xterminal;xfce4-terminal;
+X-XFCE-Binaries=xfce4-terminal;Terminal;xterminal;
 X-XFCE-Category=TerminalEmulator
 X-XFCE-Commands=%B;
 X-XFCE-CommandsWithParameter=%B -x %s;
diff --git a/exo-helper/helpers/chromium.desktop.in.in b/exo-helper/helpers/chromium.desktop.in.in
index a9edf06..b1d284f 100644
--- a/exo-helper/helpers/chromium.desktop.in.in
+++ b/exo-helper/helpers/chromium.desktop.in.in
@@ -5,7 +5,7 @@ Icon=chromium
 Type=X-XFCE-Helper
 _Name=Chromium
 StartupNotify=true
-X-XFCE-Binaries=chromium;google-chrome;
+X-XFCE-Binaries=chromium;google-chrome;chromium-browser;
 X-XFCE-Category=WebBrowser
 X-XFCE-Commands=%B;
 X-XFCE-CommandsWithParameter=%B "%s";
