--- monodevelop-vala/monodevelop-vala-2.2/configure.in	2009-12-11 21:51:36.000000000 +0100
+++ monodevelop-vala/monodevelop-vala-2.2/configure-copy.in	2010-04-25 22:58:10.000000000 +0200
@@ -82,16 +82,13 @@
 AC_SUBST(GCONF_SHARP_LIBS)
 AM_CONDITIONAL(HAVE_GCONF_SHARP, test x$gconf_sharp = xyes)
 
-AC_CHECK_PROG(VSCSHELL, vsc-shell, [vsc-shell])
-AC_SUBST(VSCSHELL)
-
-if test x$VSCSHELL = x; then
+AC_SEARCH_LIBS(afrodite_utils_get_package_paths, afrodite, [], [
 	echo ""
-	echo "Vsc-shell is required for Vala code completion."
-	echo "Vsc-shell is part of the VTG package, available from: "
+	echo "libafrodite is required for Vala code completion."
+	echo "libafrodite is part of the VTG package, available from: "
 	echo "http://code.google.com/p/vtg/"
 	echo ""
-fi
+])
 
 gtksharp_prefix="`$PKG_CONFIG --variable=prefix gtk-sharp-2.0`"
 AC_SUBST(gtksharp_prefix)
