From 455f792e4791e037eb0406005b92c345cf0baf56 Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay.sievers@vrfy.org>
Date: Wed, 10 Mar 2010 14:43:07 +0100
Subject: [PATCH 22/36] replace utimes() with utimensat()

---
 udev/udev-node.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/udev/udev-node.c b/udev/udev-node.c
index a70c147..307a76b 100644
--- a/udev/udev-node.c
+++ b/udev/udev-node.c
@@ -57,7 +57,7 @@ int udev_node_mknod(struct udev_device *dev, const char *file, dev_t devnum, mod
 			preserve = 1;
 			udev_selinux_lsetfilecon(udev, file, mode);
 			/* always update timestamp when we re-use the node, like on media change events */
-			utimes(file, NULL);
+			utimensat(AT_FDCWD, file, NULL, 0);
 		} else {
 			char file_tmp[UTIL_PATH_SIZE + sizeof(TMP_FILE_EXT)];
 
@@ -178,7 +178,7 @@ static int node_symlink(struct udev *udev, const char *node, const char *slink)
 					info(udev, "preserve already existing symlink '%s' to '%s'\n",
 					     slink, target);
 					udev_selinux_lsetfilecon(udev, slink, S_IFLNK);
-					lutimes(slink, NULL);
+					utimensat(AT_FDCWD, slink, NULL, AT_SYMLINK_NOFOLLOW);
 					goto exit;
 				}
 			}
-- 
1.6.6

