diff -rupN hydrogen-0.9.5.1.old/Sconstruct hydrogen-0.9.5.1/Sconstruct
--- hydrogen-0.9.5.1.old/Sconstruct	2012-02-19 16:54:05.000000000 -0500
+++ hydrogen-0.9.5.1/Sconstruct	2012-07-22 21:10:17.044739638 -0400
@@ -193,6 +193,8 @@ def get_hydrogen_lib( opts ):
             env.ParseConfig('pkg-config --modversion jack', get_jack_api_flags)
         if str(env['lash']) == "1":
             env.ParseConfig('pkg-config --cflags --libs lash-1.0')
+        if str(env['lrdf']) == "1":
+            env.ParseConfig('pkg-config --cflags --libs lrdf')
 
 
 	#env.Decider is not known in older scons version
@@ -450,7 +452,7 @@ elif not str(env['libarchive']) == "1" a
    Exit(1)
 
 #lrdf: categorizing of ladspa effects
-if str(env['lrdf']) == "1" and not conf.CheckCHeader('lrdf.h'):
+if str(env['lrdf']) == "1" and not conf.CheckPKG('lrdf'):
     print 'lrdf must be installed!'
     Exit(1)
 
