diff -r ac1f19d8f817 src/mainWindow.py
--- a/src/mainWindow.py	Fri Feb 29 01:56:45 2008 -0500
+++ b/src/mainWindow.py	Fri Feb 29 06:59:39 2008 -0500
@@ -30,7 +30,6 @@ import sys
 import sys
 import time
 import os
-import rpm
 
 import libuser
 
@@ -166,7 +165,7 @@ class mainWindow:
 
     def version():
         # substituted to the real version by the Makefile at installation time.
-        return "@VERSION@"
+        return "1.2.76"
 
     def destroy (self, *args):
         try:
@@ -629,7 +628,7 @@ class mainWindow:
 
         aboutDialog = xml.get_widget ("aboutDialog")
         aboutDialog.set_program_name (_("system-config-users"))
-        aboutDialog.set_version ("@VERSION@")
+        aboutDialog.set_version ("1.2.76")
         aboutDialog.set_logo (iconPixbuf)
 
         cb_strings = []
@@ -705,26 +704,8 @@ class mainWindow:
                     dlg.run ()
                     dlg.destroy ()
                     return
-                ts = rpm.TransactionSet ()
-                if ts.dbMatch ("basenames", homeDir).count () > 0 or ts.dbMatch ("basenames", os.path.abspath (homeDir)).count () > 0 or ts.dbMatch ("basenames", os.path.realpath (homeDir)).count () > 0:
-                    errMsgs.append (_("- An installed software package contains this directory."))
-                if uid < 500 or (user == "nfsnobody" and (long (uid) == 65534L or long (uid) == 4294967294L)):
-                    errMsgs.append (_("- A system user owns this directory and removing it may impair the system's integrity."))
-                if not os.access(homeDir, os.W_OK):
-                    errMsgs.append (_("- This directory doesn't exist or isn't writable."))
-                elif os.lstat (homeDir).st_uid != uid:
-                    errMsgs.append (_("- The user '%s' doesn't own this directory.") % (user))
                 
-                pipe = os.popen ("/usr/bin/pgrep -fl -U %d; /usr/bin/pgrep -u %d" % (uid, uid))
-                processes = pipe.readlines ()
-                pipe.close ()
-                processes_running_text = ""
-                if len (processes) > 0:
-                    processes_running_text = _("<b>There are currently processes running that are owned by '%s'!</b>  This user is probably still logged in.  ") % (user)
-                
-                text = _("Do you really want to remove the user '%s'?") % (user)
-                
-                dlg = gtk.MessageDialog(None, 0, gtk.MESSAGE_QUESTION, gtk.BUTTONS_YES_NO, (processes_running_text + text))
+                dlg = gtk.MessageDialog(None, 0, gtk.MESSAGE_QUESTION, gtk.BUTTONS_YES_NO, ("Are you sure you want to delete the user '%s'?" % user))
                 dlg.set_position(gtk.WIN_POS_CENTER)
                 dlg.set_modal(True)
                 dlg.set_icon(iconPixbuf)
@@ -953,11 +934,6 @@ class mainWindow:
         self.on_properties_activate()
 
     def isSELinuxInstalled(self):
-        ts = rpm.TransactionSet()
-
-        mi = ts.dbMatch('name', 'policy-sources')
-        if mi.count() > 0:
-            return True
         return False
 
     def isSELinuxEnabled(self):
