--- pam/gkr-pam-module.c.orig	2010-06-13 00:33:24.000000000 +0200
+++ pam/gkr-pam-module.c	2010-08-07 12:19:53.000000000 +0200
@@ -509,8 +509,13 @@ start_daemon (pam_handle_t *ph, struct p
 	 * for a password. Closing input signifies password is done.
 	 */
 
-	if (password)
+	/* syslog (GKR_LOG_INFO, 
+	   "gkr-pam: start_daemon() -- do we have a password for the login key ring?");
+	*/
+	if (password) {
+		syslog (GKR_LOG_INFO, "gkr-pam: start_daemon() -- writing login keyring password");
 		write_string (inp[WRITE_END], password);
+	}
 	close (inp[WRITE_END]);
 
 	/* 
@@ -584,10 +589,14 @@ start_daemon_if_necessary (pam_handle_t 
 		}
 		
 		/* Daemon is already running */
+		syslog (GKR_LOG_INFO, 
+			"gkr-pam: start_daemon_if_necessary() -- daemon is already running, transferred env variables successfully");
 		return PAM_SUCCESS;
 	}
 
 	/* Not running, start process */
+	syslog (GKR_LOG_INFO,
+		"gkr-pam: start_daemon_if_necessary() -- daemon is not running, start it now");
 	ret = start_daemon (ph, pwd, password);
 	*started = (ret == PAM_SUCCESS);
 	return ret;
@@ -664,7 +673,7 @@ unlock_keyring (pam_handle_t *ph, struct
 		return PAM_SERVICE_ERR;
 	}
 
-	syslog (GKR_LOG_INFO, "gkr-pam: unlocked login keyring");
+	syslog (GKR_LOG_INFO, "gkr-pam: unlock_keyring() --  unlocked login keyring");
 	return PAM_SUCCESS;
 }
 
@@ -808,32 +817,49 @@ pam_sm_authenticate (pam_handle_t *ph, i
 	/* Figure out and/or prompt for the user name */
 	ret = pam_get_user (ph, &user, NULL);
 	if (ret != PAM_SUCCESS) {
-		syslog (GKR_LOG_ERR, "gkr-pam: couldn't get the user name: %s", 
+		syslog (GKR_LOG_ERR, "gkr-pam: (auth) couldn't get the user name: %s", 
 		        pam_strerror (ph, ret));
 		return PAM_SERVICE_ERR;
 	}
 	
 	pwd = getpwnam (user);
 	if (!pwd) {
-		syslog (GKR_LOG_ERR, "gkr-pam: error looking up user information for: %s", user);
+		syslog (GKR_LOG_ERR, "gkr-pam: (auth) error looking up user information for: %s", user);
 		return PAM_SERVICE_ERR;
 	}
 		
 	/* Look up the password */
 	ret = pam_get_item (ph, PAM_AUTHTOK, (const void**)&password);
 	if (ret != PAM_SUCCESS || password == NULL) {
-		if (ret == PAM_SUCCESS)
-			syslog (GKR_LOG_WARN, "gkr-pam: no password is available for user");
-		else
-			syslog (GKR_LOG_WARN, "gkr-pam: no password is available for user: %s", 
+		if (args & ARG_USE_AUTHTOK) {
+			if (ret == PAM_SUCCESS)
+				syslog (GKR_LOG_WARN, "gkr-pam: (auth) no password is available for user");
+			else
+				syslog (GKR_LOG_WARN, "gkr-pam: (auth) no password is available for user: %s",
+					pam_strerror (ph, ret));
+			return PAM_SUCCESS;
+		}
+
+		ret = prompt_password (ph);
+		if (ret != PAM_SUCCESS) {
+			syslog (GKR_LOG_ERR, "gkr-pam: (auth) couldn't get the password from user: %s",
 			        pam_strerror (ph, ret));
-		return PAM_SUCCESS;
+			return PAM_AUTH_ERR;
+		}
+		ret = pam_get_item (ph, PAM_AUTHTOK, (const void**)&password);
+		if (ret != PAM_SUCCESS || password == NULL) {
+			syslog (GKR_LOG_ERR, "gkr-pam: (auth) couldn't get the password from user: %s",
+				ret == PAM_SUCCESS ? "password was null" : pam_strerror (ph, ret));
+			return PAM_AUTH_ERR;
+		}
 	}
+	syslog (GKR_LOG_INFO, "gkr-pam: (auth) successfully looked up password for %s", user);
 
 	started_daemon = 0;
 
 	/* Should we start the daemon? */
 	if (args & ARG_AUTO_START) {
+		/* syslog (GKR_LOG_INFO, "gkr-pam: (auth) calling start_daemon_if_necessary()"); */
 		ret = start_daemon_if_necessary (ph, pwd, password, &started_daemon);
 		if (ret != PAM_SUCCESS)
 			return ret;
@@ -848,15 +874,19 @@ pam_sm_authenticate (pam_handle_t *ph, i
 			ret = unlock_keyring (ph, pwd, password);
 			if (ret != PAM_SUCCESS)
 				return ret;
+			syslog (GKR_LOG_INFO,
+				"gkr-pam: (auth) daemon already running, successfully unlocked login keyring");
 		}
 		
 	/* Otherwise start later in open session, store password */
 	} else {
 		if (pam_set_data (ph, "gkr_system_authtok", strdup (password),
 		                  cleanup_free_password) != PAM_SUCCESS) {
-			syslog (GKR_LOG_ERR, "gkr-pam: error storing authtok");
+			syslog (GKR_LOG_ERR, "gkr-pam: (auth) error storing authtok");
 			return PAM_AUTHTOK_RECOVER_ERR;
 		}
+		syslog (GKR_LOG_INFO, 
+			"gkr-pam: (auth) not starting daemon now, successfully stored authtok for later use in (session)");
  	}
 
 	return PAM_SUCCESS;
@@ -876,20 +906,25 @@ pam_sm_open_session (pam_handle_t *ph, i
 	if (args & ARG_IGNORE_SERVICE)
 		return PAM_SUCCESS;
 
+	/* syslog (GKR_LOG_INFO, "gkr-pam: (session) checkpoint #1"); */
 	/* Figure out the user name */
 	ret = pam_get_user (ph, &user, NULL);
 	if (ret != PAM_SUCCESS) {
-		syslog (GKR_LOG_ERR, "gkr-pam: couldn't get the user name: %s", 
+		syslog (GKR_LOG_ERR, "gkr-pam: (session) couldn't get the user name: %s", 
 		        pam_strerror (ph, ret));
 		return PAM_SERVICE_ERR;
 	}
 
+	/* syslog (GKR_LOG_INFO, "gkr-pam: (session) checkpoint #2 -- username found"); */
+
 	pwd = getpwnam (user);
 	if (!pwd) {
-		syslog (GKR_LOG_ERR, "gkr-pam: error looking up user information for: %s", user);
+		syslog (GKR_LOG_ERR, "gkr-pam: (session) error looking up user information for: %s", user);
 		return PAM_SERVICE_ERR;
 	}
 
+	/* syslog (GKR_LOG_INFO, "gkr-pam: (session) checkpoint #3 -- user information found"); */
+
 	/* Get the stored authtok here */
 	if (pam_get_data (ph, "gkr_system_authtok", (const void**)&password) != PAM_SUCCESS) {
 		/* 
@@ -900,12 +935,15 @@ pam_sm_open_session (pam_handle_t *ph, i
 		 * No use complaining
 		 */
 		password = NULL;
+		syslog (GKR_LOG_INFO, "gkr-pam: (session) error retrieving stored authtok");
 	}
+	/* syslog (GKR_LOG_INFO, "gkr-pam: (session) checkpoint #4 -- retrieved stored authtok"); */
 	
 	started_daemon = 0;
 	
 	/* Should we start the daemon? */
 	if (args & ARG_AUTO_START) {
+		/* syslog (GKR_LOG_INFO, "gkr-pam: (session) calling start_daemon_if_necessary()"); */
 		ret = start_daemon_if_necessary (ph, pwd, password, &started_daemon);
 		if (ret != PAM_SUCCESS)
 			return ret;
@@ -914,10 +952,16 @@ pam_sm_open_session (pam_handle_t *ph, i
 	/* If gnome keyring is running, but we didn't start it here, then unlock now */
 	if (get_any_env (ph, ENV_CONTROL) != NULL) {
 		if (!started_daemon && password != NULL) {
-			if (unlock_keyring (ph, pwd, password) != PAM_SUCCESS)
+			if (unlock_keyring (ph, pwd, password) != PAM_SUCCESS) {
+				syslog (GKR_LOG_INFO , 
+					"gkr-pam: (session) daemon already running (started elsewhere), but login keyring cannot be unlocked (and password is !NULL");
 				return PAM_SERVICE_ERR;
+			}
+			syslog (GKR_LOG_INFO, 
+				"gkr-pam: (session) daemon already running (started elsewhere), successfully unlocked login keyring");
 		}
 	}
+	/* syslog (GKR_LOG_INFO, "gkr-pam: (session) checkpoint #5 (see daemon info above)"); */
 	
 	return PAM_SUCCESS;
 }
