--- configure	2009-03-18 19:45:27.000000000 +0800
+++ configure-new	2009-08-04 16:30:13.000000000 +0800
@@ -2362,7 +2362,7 @@
 
 
 
-GACUTIL_FLAGS='/package $(PACKAGE) /gacdir $(DESTDIR)/$(prefix)/lib'
+GACUTIL_FLAGS='/package $(PACKAGE) /gacdir $(DESTDIR)/$(libdir)'
 
 
 ac_config_files="$ac_config_files Makefile bin/Makefile extras/Makefile extras/booc extras/booi extras/booish extras/boo.pc"
--- bin/Makefile.in	2009-08-04 16:33:06.000000000 +0800
+++ bin/Makefile.in-new	2009-08-04 16:33:27.000000000 +0800
@@ -127,7 +127,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-boodir = $(prefix)/lib/boo
+boodir = $(libdir)/boo
 boo_DATA = booc.exe booi.exe booish.exe booc.rsp booc.exe.config booi.exe.config booish.exe.config
 BOO_GAC_DLLS = Boo.Lang.dll \
                Boo.Lang.Useful.dll \
--- extras/Makefile.in	2009-03-18 19:45:28.000000000 +0800
+++ extras/Makefile.in-new	2009-08-04 16:34:57.000000000 +0800
@@ -136,7 +136,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-pkgconfigdir = $(prefix)/lib/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = boo.pc
 gtksourceviewdir = $(GTKSOURCEVIEW_PREFIX)/share/gtksourceview-1.0/language-specs/
 gtksourceview_DATA = boo.lang
--- extras/boo.pc.in    2009-03-18 19:38:07.000000000 +0800
+++ extras/boo.pc.in-new        2009-08-04 17:17:08.000000000 +0800
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=${prefix}/lib
+libdir=@libdir@/mono
 
 
 Name: Boo
--- extras/booi.in	2009-03-18 19:38:07.000000000 +0800
+++ extras/booi.in-new	2009-08-04 16:41:09.000000000 +0800
@@ -1,7 +1,7 @@
 #!/bin/sh
 if [ -x @prefix@/bin/cli ]; then
-    env @prefix@/bin/cli $MONO_OPTIONS @prefix@/lib/boo/booi.exe $BOOI_OPTIONS "$@"
+    env @prefix@/bin/cli $MONO_OPTIONS @libdir@/boo/booi.exe $BOOI_OPTIONS "$@"
 else
-    env @RUNTIME@ $MONO_OPTIONS @prefix@/lib/boo/booi.exe $BOOI_OPTIONS "$@"
+    env @RUNTIME@ $MONO_OPTIONS @libdir@/boo/booi.exe $BOOI_OPTIONS "$@"
 fi
 
--- extras/booc.in	2009-03-18 19:38:07.000000000 +0800
+++ extras/booc.in-new	2009-08-04 16:43:22.000000000 +0800
@@ -1,7 +1,7 @@
 #!/bin/sh
 if [ -x @prefix@/bin/cli ]; then
-    env @prefix@/bin/cli $MONO_OPTIONS @prefix@/lib/boo/booc.exe $BOOC_OPTIONS "$@"
+    env @prefix@/bin/cli $MONO_OPTIONS @libdir@/boo/booc.exe $BOOC_OPTIONS "$@"
 else
-    env @RUNTIME@ $MONO_OPTIONS @prefix@/lib/boo/booc.exe $BOOC_OPTIONS "$@"
+    env @RUNTIME@ $MONO_OPTIONS @libdir@/boo/booc.exe $BOOC_OPTIONS "$@"
 fi
 
--- extras/booish.in	2009-03-18 19:38:07.000000000 +0800
+++ extras/booish.in-new	2009-08-04 16:44:35.000000000 +0800
@@ -1,7 +1,7 @@
 #!/bin/sh
 if [ -x @prefix@/bin/cli ]; then
-    env @prefix@/bin/cli $MONO_OPTIONS @prefix@/lib/boo/booish.exe $BOOISH_OPTIONS "$@"
+    env @prefix@/bin/cli $MONO_OPTIONS @libdir@/boo/booish.exe $BOOISH_OPTIONS "$@"
 else
-    env @RUNTIME@ $MONO_OPTIONS @prefix@/lib/boo/booish.exe $BOOISH_OPTIONS "$@"
+    env @RUNTIME@ $MONO_OPTIONS @libdir@/boo/booish.exe $BOOISH_OPTIONS "$@"
 fi
 
