From 5225ebf3660a7c31242f03ee4fb29f377c69f52b Mon Sep 17 00:00:00 2001
From: Guido Berhoerster <guido+xfce@berhoerster.name>
Date: Thu, 11 Oct 2012 21:18:57 +0000
Subject: Initialize window width and height to default values

---
diff --git a/libxfce4mixer/xfce-mixer-preferences.c b/libxfce4mixer/xfce-mixer-preferences.c
index 117ebcf..a5e5996 100644
--- a/libxfce4mixer/xfce-mixer-preferences.c
+++ b/libxfce4mixer/xfce-mixer-preferences.c
@@ -134,8 +134,8 @@ xfce_mixer_preferences_init (XfceMixerPreferences *preferences)
 {
   preferences->channel = xfconf_channel_get ("xfce4-mixer");
 
-  preferences->window_width = 1;
-  preferences->window_height = 1;
+  preferences->window_width = 640;
+  preferences->window_height = 400;
 
   preferences->sound_card = NULL;
 
--
cgit v0.9.0.3
