Index: kdebase-runtime-4.5.2/knetattach/knetattach.desktop
===================================================================
--- kdebase-runtime-4.5.2.orig/knetattach/knetattach.desktop	2010-10-16 06:14:57.685310014 +0200
+++ kdebase-runtime-4.5.2/knetattach/knetattach.desktop	2010-10-16 06:15:28.921310006 +0200
@@ -180,3 +180,4 @@
 Categories=Qt;KDE;Network;RemoteAccess;
 OnlyShowIn=KDE;
 X-DocPath=knetattach/index.html
+NoDisplay=true
