diff -rupN ardour-2.8.old/libs/ardour/session.cc ardour-2.8/libs/ardour/session.cc
--- ardour-2.8.old/libs/ardour/session.cc	2009-03-26 17:57:25.000000000 -0400
+++ ardour-2.8/libs/ardour/session.cc	2009-03-30 14:59:10.000000000 -0400
@@ -3886,7 +3886,8 @@ Session::ensure_passthru_buffers (uint32
 #endif			
 		_passthru_buffers.push_back (p);
 
-		*p = 0;
+		if (current_block_size > 0)
+		  *p = 0;
 		
 #ifdef NO_POSIX_MEMALIGN
 		p =  (Sample *) malloc(current_block_size * sizeof(Sample));
@@ -3901,7 +3902,8 @@ Session::ensure_passthru_buffers (uint32
 		memset (p, 0, sizeof (Sample) * current_block_size);
 		_silent_buffers.push_back (p);
 
-		*p = 0;
+		if (current_block_size > 0)
+		  *p = 0;
 		
 #ifdef NO_POSIX_MEMALIGN
 		p =  (Sample *) malloc(current_block_size * sizeof(Sample));

