
https://bugzilla.redhat.com/show_bug.cgi?id=379811

--- httpd-2.2.3/modules/cache/mod_cache.c.cachehardmax
+++ httpd-2.2.3/modules/cache/mod_cache.c
@@ -736,6 +736,11 @@ static int cache_save_filter(ap_filter_t
             exp = date + conf->defex;
         }
     }
+    /* else, forcibly cap the expiry date if required */
+    else if (conf->hardmaxex && (date + conf->maxex) < exp) {
+        exp = date + conf->maxex;
+    }        
+
     info->expire = exp;
 
     /* We found a stale entry which wasn't really stale. */
@@ -926,6 +931,9 @@ static void * create_cache_config(apr_po
     /* flag indicating that query-string should be ignored when caching */
     ps->ignorequerystring = 0;
     ps->ignorequerystring_set = 0;
+    /* flag; treat maxex as hard limit */
+    ps->hardmaxex = 0;
+    ps->hardmaxex_set = 0;
     return ps;
 }
 
@@ -975,6 +983,12 @@ static void * merge_cache_config(apr_poo
         (overrides->ignorequerystring_set == 0)
         ? base->ignorequerystring
         : overrides->ignorequerystring;
+
+    ps->hardmaxex = 
+        (overrides->hardmaxex_set == 0)
+        ? base->hardmaxex
+        : overrides->hardmaxex;
+
     return ps;
 }
 static const char *set_cache_ignore_no_last_mod(cmd_parms *parms, void *dummy,
@@ -1111,7 +1125,7 @@ static const char *add_cache_disable(cmd
 }
 
 static const char *set_cache_maxex(cmd_parms *parms, void *dummy,
-                                   const char *arg)
+                                   const char *arg, const char *hard)
 {
     cache_server_conf *conf;
 
@@ -1120,6 +1134,12 @@ static const char *set_cache_maxex(cmd_p
                                                   &cache_module);
     conf->maxex = (apr_time_t) (atol(arg) * MSEC_ONE_SEC);
     conf->maxex_set = 1;
+    
+    if (hard && strcasecmp(hard, "hard") == 0) {
+        conf->hardmaxex = 1;
+        conf->hardmaxex_set = 1;
+    }
+
     return NULL;
 }
 
@@ -1196,7 +1216,7 @@ static const command_rec cache_cmds[] =
                   "caching is enabled"),
     AP_INIT_TAKE1("CacheDisable", add_cache_disable, NULL, RSRC_CONF,
                   "A partial URL prefix below which caching is disabled"),
-    AP_INIT_TAKE1("CacheMaxExpire", set_cache_maxex, NULL, RSRC_CONF,
+    AP_INIT_TAKE12("CacheMaxExpire", set_cache_maxex, NULL, RSRC_CONF,
                   "The maximum time in seconds to cache a document"),
     AP_INIT_TAKE1("CacheDefaultExpire", set_cache_defex, NULL, RSRC_CONF,
                   "The default time in seconds to cache a document"),
--- httpd-2.2.3/modules/cache/mod_cache.h.cachehardmax
+++ httpd-2.2.3/modules/cache/mod_cache.h
@@ -153,6 +153,9 @@ typedef struct {
     /** ignore query-string when caching */
     int ignorequerystring;
     int ignorequerystring_set;
+    /* treat maxex as hard limit */
+    int hardmaxex;
+    int hardmaxex_set;
 } cache_server_conf;
 
 /* cache info information */
