diff -Naur libftdi-0.17/bindings/python/setup.py.in libftdi-0.17-mod/bindings/python/setup.py.in
--- libftdi-0.17/bindings/python/setup.py.in	2009-05-29 22:54:41.000000000 +0300
+++ libftdi-0.17-mod/bindings/python/setup.py.in	2010-01-04 09:48:58.000000000 +0200
@@ -12,6 +12,6 @@
       py_modules=['ftdi'],
       ext_modules=[Extension('_ftdi', ['ftdi_wrap.c'],
                              include_dirs=['@top_srcdir@/src'],
-                             library_dirs=['@top_buildir@/src/.libs','lib','/usr/lib'],
+                             library_dirs=['@top_builddir@/src/.libs','lib','/usr/lib'],
                              libraries=['ftdi'])],
       )
