--- vnc-4_1_2-unixsrc/unix/xorg-server-1.4/hw/vnc/Makefile.am.jau	2007-11-14 14:15:03.000000000 -0500
+++ vnc-4_1_2-unixsrc/unix/xorg-server-1.4/hw/vnc/Makefile.am	2007-11-14 14:17:13.000000000 -0500
@@ -24,22 +24,11 @@
 		$(top_srcdir)/Xext/dpmsstubs.c \
 		$(top_srcdir)/Xi/stubs.c \
 		$(top_srcdir)/mi/miinitext.c \
-		$(top_srcdir)/fb/fbcmap.c
+		$(top_srcdir)/fb/fbcmap_mi.c
 
 Xvnc_LDADD =	$(XORG_CORE_LIBS) \
-		$(XORG_LIBS) \
+		$(XVFB_LIBS) \
 		$(XSERVER_LIBS) \
-		../../fb/libfb.la \
-		../../mi/libminimi.la \
-		../../render/librender.la \
-		../../xkb/libxkbstubs.la \
-		../../dbe/libdbe.la \
-		../../Xext/libXext.la \
-		../../XTrap/libxtrap.la \
-		../../record/librecord.la \
-		../xfree86/os-support/libxorgos.la \
-		../../os/libos.la \
-		../xfree86/common/libcommon.la \
 		-lX11 \
 		libvnccommon.la \
 		../../../../common/rfb/librfb.la
