diff -up sysstat-9.0.6.1/configure.in.ppp sysstat-9.0.6.1/configure.in
--- sysstat-9.0.6.1/configure.in.ppp	2009-11-21 06:55:46.000000000 +0100
+++ sysstat-9.0.6.1/configure.in	2010-09-21 10:01:55.000000000 +0200
@@ -336,10 +336,9 @@ AC_MSG_CHECKING(whether man_group variab
 AC_ARG_ENABLE(man-group,
 	      AC_HELP_STRING([--disable-man-group],
 			     [ignore man_group variable value]),
-			     AUX_IMG=$enableval,AUX_IMG=no)
+			     AUX_IMG=yes,AUX_IMG=no)
 if test $AUX_IMG != "yes"; then
    IGNORE_MAN_GROUP=n
-   AUX_IMG=no
 else
    IGNORE_MAN_GROUP=y
 fi
