diff -r 1b26e4b4cbb1 conary/lib/Makefile
--- a/conary/lib/Makefile	Wed Aug 08 13:09:32 2012 +0100
+++ b/conary/lib/Makefile	Wed Aug 22 16:22:37 2012 +0100
@@ -26,10 +26,6 @@
 libelf = $(shell \
 if [ -x /usr/bin/pkg-config ] && /usr/bin/pkg-config --exists libelf; then \
     pkg-config --cflags --libs libelf; \
-elif [ -d /usr/include/libelf-lgpl ]; then \
-    echo "-I/usr/include/libelf-lgpl -lelf-lgpl"; \
-elif [ -d ${SYSROOT}/usr/include/libelf-lgpl ]; then \
-    echo "-I${SYSROOT}/usr/include/libelf-lgpl -lelf-lgpl"; \
 elif [ -d /usr/include/libelf ] && grep 'GNU \(Library\|Lesser\) General Public' /usr/include/libelf/libelf.h >/dev/null; then\
     echo "-I/usr/include/libelf -lelf"; \
 elif [ -d ${SYSROOT}/usr/include/libelf ] && grep 'GNU \(Library\|Lesser\) General Public' ${SYSROOT}/usr/include/libelf/libelf.h >/dev/null; then\
