diff -r 05d29b2d124b schemas/desktop_gnome_url_handlers.schemas
--- a/schemas/desktop_gnome_url_handlers.schemas	Wed Mar 19 10:03:42 2008 -0400
+++ b/schemas/desktop_gnome_url_handlers.schemas	Wed Mar 19 10:04:51 2008 -0400
@@ -4503,7 +4503,7 @@
       <applyto>/desktop/gnome/url-handlers/http/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "http" URLs</short>
         <long>The command used to handle "http" URLs, if enabled.</long>
@@ -5532,7 +5532,7 @@
       <applyto>/desktop/gnome/url-handlers/https/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "https" URLs</short>
         <long>The command used to handle "https" URLs, if enabled.</long>
diff -r 05d29b2d124b schemas/desktop_gnome_url_handlers.schemas.in
--- a/schemas/desktop_gnome_url_handlers.schemas.in	Wed Mar 19 10:03:42 2008 -0400
+++ b/schemas/desktop_gnome_url_handlers.schemas.in	Wed Mar 19 10:04:51 2008 -0400
@@ -153,7 +153,7 @@
       <applyto>/desktop/gnome/url-handlers/http/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "http" URLs</short>
         <long>The command used to handle "http" URLs, if enabled.</long>
@@ -187,7 +187,7 @@
       <applyto>/desktop/gnome/url-handlers/https/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "https" URLs</short>
         <long>The command used to handle "https" URLs, if enabled.</long>
