diff -up ./configure.ac.old ./configure.ac
--- ./configure.ac.old	2007-12-29 14:14:54.000000000 -0800
+++ ./configure.ac	2007-12-29 14:15:15.000000000 -0800
@@ -30,8 +30,6 @@ GNOME_DOC_INIT
 AM_GCONF_SOURCE_2
 
 AM_PATH_PYTHON([2.5])
-AC_LANG([C])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([[#include <stdlib.h>]],[[int result = system("python depcheck.py"); if (result != 0) exit(EXIT_FAILURE);]])], [], [AC_MSG_ERROR([Error: Dependency check failed])])
 
 AC_CONFIG_FILES([
 po/Makefile.in
