diff -ur farsight-0.1.26.OLD/farsight/transmitters/libjingle/jingle_c.cpp farsight-0.1.26/farsight/transmitters/libjingle/jingle_c.cpp
--- farsight-0.1.26.OLD/farsight/transmitters/libjingle/jingle_c.cpp	2008-03-27 20:49:18.000000000 -0400
+++ farsight-0.1.26/farsight/transmitters/libjingle/jingle_c.cpp	2008-03-27 20:50:04.000000000 -0400
@@ -29,6 +29,7 @@
 #include <talk/base/physicalsocketserver.h>
 #include <talk/base/socketaddress.h>
 #include <talk/p2p/client/socketclient.h>
+#include <string.h>
 
 class SignalListener1 : public sigslot::has_slots<>
 {
